backend/tests/e2e/test_api_e2e.py

```
@@ -453,247 +671,209 @@
+        # SANITY CHECK: Duration should be non-negative (exact value depends on execution speed)
+        duration = summary_data.get("total_duration_ms", -1)
+        assert duration >= 0, f"SANITY CHECK FAILED: summary endpoint shows negative processing duration ({duration}ms)"
+        
+        # Phase 5: Chain statistics validation - now nested in SessionStats
+        assert "chain_statistics" in summary_data, "Chain statistics missing from summary endpoint"
+        chain_stats = summary_data["chain_statistics"]
+        chain_fields = ["total_stages", "completed_stages", "failed_stages"]
+        for field in chain_fields:
+            assert field in chain_stats, f"Chain statistics missing field: {field}"
         
         print(f"✅ Summary endpoint validation passed - data consistency confirmed")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)